### PR TITLE
Self-contained shoot-item mixin

### PR DESCRIPTION
**Pull request by holgerkoser**
**What this PR does / why we need it**:
The `shoot-item` mixin requires that the component that declares the mixin has a property with the name `shootItem`. This property should be defined in the mixin itself. This PR moves the `shootItem` property definition from the components into the mixin.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```bugfix user

```

### CODE REVIEW

File: frontend/src/components/ShootDetails/ShootExternalToolsCard.vue [hunk 1]
```diff
@@ -10,11 +10,11 @@ SPDX-License-Identifier: Apache-2.0
       <v-toolbar-title class="text-subtitle-1">External Tools</v-toolbar-title>
     </v-toolbar>
     <v-list>
-      <template v-for="({ title, url, icon = 'link' }, index) in items">
+      <template v-for="({ title, url, icon }, index) in items">
         <v-divider v-if="index" :key="index" inset class="my-2"></v-divider>
         <v-list-item :key="title">
           <v-list-item-icon>
-            <v-icon color="primary">{{icon}}</v-icon>
+            <v-icon color="primary">{{icon || 'link'}}</v-icon>
```
Comment (holgerkoser) [inline frontend/src/components/ShootDetails/ShootExternalToolsCard.vue:17]: VS-Code cannot handle javascript default values in templates. Syntax highlighting is broken.

File: frontend/src/components/ShootMessages/ShootMessages.vue [hunk 2]
```diff
@@ -164,7 +161,7 @@ export default {
             purposeText: this.shootPurpose || '',
             shootName: this.shootName || '',
             shootNamespace: this.shootNamespace || '',
-            showNavigationLink: this.canPatchShoots
+            showNavigationLink: this.canPatchShoots && !isEmpty(this.shootName) && !isEmpty(this.shootNamespace)
```
Comment (holgerkoser) [inline frontend/src/components/ShootMessages/ShootMessages.vue:164]: Sometimes `shootNamespace` and `shootName` is empty. Rendering the link leads to an error in this case.

File: frontend/src/views/ShootList.vue [hunk 3]
```diff
@@ -228,7 +228,6 @@ export default {
   computed: {
     ...mapGetters({
       mappedItems: 'shootList',
-      item: 'shootByNamespaceAndName',
```
Comment (holgerkoser) [inline frontend/src/views/ShootList.vue:231]: Unused?

File: frontend/src/views/ShootItemTerminal.vue [hunk 4]
```diff
@@ -18,16 +18,12 @@ SPDX-License-Identifier: Apache-2.0
...
 import get from 'lodash/get'
 import TerminalSplitpanes from '@/components/TerminalSplitpanes'
-import { mapGetters } from 'vuex'
 
 export default {
   components: {
     TerminalSplitpanes
   },
   computed: {
-    ...mapGetters([
```
Comment (holgerkoser) [inline frontend/src/views/ShootItemTerminal.vue:28]: unused?

File: frontend/src/components/dialogs/CreateTerminalSessionDialog.vue [hunk 5]
```diff
@@ -103,13 +102,9 @@ export default {
     UnverifiedTerminalShortcutsDialog,
     WebterminalServiceAccountDialog
   },
-  mixins: [shootItem],
```
Comment (holgerkoser) [inline frontend/src/components/dialogs/CreateTerminalSessionDialog.vue:106]: Mixin was not used at all?

File: frontend/src/components/TerminalTarget.vue [hunk 6]
```diff
@@ -69,14 +72,19 @@ export default {
...
       'hasShootTerminalAccess',
       'hasGardenTerminalAccess',
       'isAdmin'
     ]),
+    shootName () {
+      return get(this.shootItem, 'metadata.name')
+    },
+    isShootStatusHibernated () {
+      return isShootStatusHibernated(get(this.shootItem, 'status'))
+    },
```
Comment (holgerkoser) [inline frontend/src/components/TerminalTarget.vue:87]: The only two methods used from the mixin. The `shootItem` property can be empty in this case. In the mixin I have defined it as required.

File: frontend/src/views/ShootItem.vue [hunk 7]
```diff
@@ -31,25 +31,25 @@ import TerminalSplitpanes from '@/components/TerminalSplitpanes'
...
     ...mapGetters([
       'shootByNamespaceAndName'
     ]),
-    value () {
-      return this.shootByNamespaceAndName(this.$route.params)
+    shootName () {
+      return get(this.$route.params, 'name')
+    },
+    shootNamespace () {
+      return get(this.$route.params, 'namespace')
```
Comment (holgerkoser) [inline frontend/src/views/ShootItem.vue:49]: The only two methods used from the mixin?

**Review by grolu — APPROVED** [review]
/lgtm

**Pull request merged**